src/python/gudhi/sklearn/cover_complex.py

```diff
@@ -0,0 +1,595 @@
+# See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+# Author(s):       Mathieu Carrière
+#
+# Copyright (C) 2021 Inria
+#
+# Modification(s):
+#   - YYYY/MM Author: Description of the modification
+
+import numpy as np
+import itertools
+import matplotlib
+import matplotlib.pyplot as plt
+
+from sklearn.base            import BaseEstimator
+from sklearn.cluster         import DBSCAN, AgglomerativeClustering
+from sklearn.metrics         import pairwise_distances
+from scipy.spatial.distance  import directed_hausdorff
+
+from .. import SimplexTree, CoverComplex
+
+class CoverComplexPy(BaseEstimator):
+    """
+    This is a mother class for MapperComplex, GraphInducedComplex and NerveComplex.
+
+    Attributes:
+        simplex_tree_ (gudhi SimplexTree): simplicial complex representing the cover complex computed after calling the fit() method.
+        node_info_ (dictionary): various information associated to the nodes of the cover complex.
+    """
+    def __init__(self, verbose=False):
+        """
+        Constructor for the CoverComplexPy class.
+
+        Parameters:
+        """
+        self.verbose = verbose
+
+    def get_networkx(self, set_attributes_from_colors=False):
+        """
+        Turn the 1-skeleton of the cover complex computed after calling fit() method into a networkx graph.
+        This function requires networkx (https://networkx.org/documentation/stable/install.html).
+
+        Parameters:
+            get_attrs (bool): if True, the color functions will be used as attributes for the networkx graph.
+
+        Returns:
+            G (networkx graph): graph representing the 1-skeleton of the cover complex.
+        """
+        import networkx as nx
+        st = self.simplex_tree_
+        G = nx.Graph()
+        for (splx,_) in st.get_skeleton(1):	
+            if len(splx) == 1:
+                G.add_node(splx[0])
+            if len(splx) == 2:
+                G.add_edge(splx[0], splx[1])
+        if set_attributes_from_colors:
+            attrs = {k: {"attr_name": self.node_info_[k]["colors"]} for k in G.nodes()}
+            nx.set_node_attributes(G, attrs)
+        return G
+
+    def save_to_dot(self, data_name="data", cover_name="cover", color_name="color", eps_color=.1, eps_size=.1):
+        """
+        Write the 0-skeleton of the cover complex in a DOT file called "{data_name_cover_name_color_name}.dot", that can be processed with, e.g., neato. The vertices of the cover complex are colored with the first color function, ie, the first column of self.colors.  This function also produces an extra pdf file "colorbar_{color_name}.pdf" containing a colorbar corresponding to the node colors in the DOT file.
+
+        Parameters:
+            data_name (string): name to use for the data, default "data"
+            cover_name (string): name to use for the cover, default "cover"
+            color_name (string): name to use for the color, default "color"
```

Comment:
Well, I thought about it, and gave up on consistency; I think it will make things clearer. What I propose now is to only provide a filename, and the data/cover/color names are asked specifically for each save function if needed.